### PR TITLE
WIP: Added missing functions to jsre module

### PR DESCRIPTION
**Pull request by drkameleon**
I have added the following functions to the *jsre* module:

- replace
- ~~replaceAll~~
- split
- match
- startsWith
- endsWith

According to the specifications here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions

### CODE REVIEW

**Comment by Araq** [comment]
@juancarlospaco please review.

File: lib/js/jsre.nim [hunk 1]
```diff
@@ -30,6 +30,18 @@ func newRegExp*(pattern: cstring): RegExp {.importjs: "new RegExp(@)".}
 func compile*(self: RegExp; pattern: cstring; flags: cstring) {.importjs: "#.compile(@)".}
   ## Recompiles a regular expression during execution of a script.
 
+func replace*(pattern: cstring; self: RegExp; replacement: cstring): cstring {.importjs: "#.replace(#,#)".}
+  ## Returns a new string with some or all matches of a pattern replaced by given replacement
+
+# func replaceAll*(pattern: cstring; self: RegExp; replacement: cstring): cstring {.importjs: "#.replaceAll(#,#)".}
+#   ## Returns a new string with all matches of a pattern replaced by given replacement
```
Comment (juancarlospaco) [inline lib/js/jsre.nim]: Remove or fix commented out code.

File: lib/js/jsre.nim [hunk 2]
```diff
@@ -30,6 +30,18 @@ func newRegExp*(pattern: cstring): RegExp {.importjs: "new RegExp(@)".}
 func compile*(self: RegExp; pattern: cstring; flags: cstring) {.importjs: "#.compile(@)".}
   ## Recompiles a regular expression during execution of a script.
 
+func replace*(pattern: cstring; self: RegExp; replacement: cstring): cstring {.importjs: "#.replace(#,#)".}
```
Comment (juancarlospaco) [inline lib/js/jsre.nim]:
```suggestion
func replace*(pattern: cstring; self: RegExp; replacement: cstring): cstring {.importjs: "#.replace(#, #)".}
```

Do not "Minify" the generated code.

File: lib/js/jsre.nim [hunk 3]
```diff
@@ -61,3 +87,12 @@ runnableExamples:
...
   assert $jsregex == "/[0-9]/i"
+  jsregex.compile(r"abc", r"i")
+  assert "abcd".startsWith jsregex
+  assert "dabc".endsWith jsregex
+  jsregex.compile(r"\d", r"i")
+  assert "do1ne".split(jsregex) == @["do".cstring, "ne".cstring]
+  jsregex.compile(r"[lw]", r"i")
+  assert "hello world".replace(jsregex,"X") == "heXlo world"
+  # jsregex.compile(r"[lw]", r"g")
+  # assert "hello world".replaceAll(jsregex,"X") == "heXXo XorXd"
```
Comment (juancarlospaco) [inline lib/js/jsre.nim]: Fix or remove commented out code.
Must have a new line at the end of file.

**Comment by juancarlospaco** [comment]
Nice work, good to see that people find `jsre` useful, and contributing to it. :slightly_smiling_face:

File: lib/js/jsre.nim (hunk 1 above)
Comment (drkameleon) [reply lib/js/jsre.nim]: I initially thought `replaceAll` was already supported, but then I discovered it was just an ES2021 thing... so out it goes. :)

https://backbencher.dev/javascript/es2021-new-features

**Comment by drkameleon** [comment]
> Nice work, good to see that people find `jsre` useful, and contributing to it. 🙂

I've been playing with the... JavaScript side of Nim lately, a lot. And yes, jsre *is* a useful, albeit minimalistic, module.

**Pull request merged**